android/app/src/main/java/com/microsoft/codepush/common/managers/CodePushAcquisitionManager.java

```diff
@@ -0,0 +1,198 @@
+            throw new CodePushQueryUpdateException("Calling common acquisition SDK with incorrect package");
+        }
+        CodePushUpdateRequest updateRequest = CodePushUpdateRequest.createUpdateRequest(mDeploymentKey, currentPackage, mClientUniqueId);
+        try {
+            final String requestUrl = mServerUrl + String.format(Locale.getDefault(), UPDATE_CHECK_ENDPOINT, mCodePushUtils.getQueryStringFromObject(updateRequest));
+            QueryUpdateJob queryUpdateJob = new QueryUpdateJob(mFileUtils, mCodePushUtils);
+            queryUpdateJob.setParameters(requestUrl);
+            queryUpdateJob.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+            try {
+                CodePushUpdateResponse codePushUpdateResponse = queryUpdateJob.get();
+                if (codePushUpdateResponse.isFailed()) {
+                    throw codePushUpdateResponse.getCodePushQueryUpdateException();
+                }
+                CodePushUpdateResponseUpdateInfo updateInfo = codePushUpdateResponse.getUpdateInfo();
+                if (updateInfo.isUpdateAppVersion()) {
+                    return CodePushRemotePackage.createDefaultRemotePackage(updateInfo.getAppVersion(), updateInfo.isUpdateAppVersion());
+                } else if (!updateInfo.isAvailable()) {
+                    return null;
+                }
+                return CodePushRemotePackage.createRemotePackageFromUpdateInfo(mDeploymentKey, updateInfo);
+            } catch (ExecutionException | InterruptedException e) {
+                throw new CodePushQueryUpdateException(e, currentPackage.getPackageHash());
+            }
+        } catch (CodePushMalformedDataException e) {
+            throw new CodePushQueryUpdateException(e, currentPackage.getPackageHash());
+        }
+    }
+
+    /**
+     * Sends deployment status to server.
+     *
+     * @param deploymentStatusReport instance of {@link CodePushDeploymentStatusReport}.
+     * @throws CodePushReportStatusException exception occurred when sending the status.
+     */
+    public void reportStatusDeploy(CodePushDeploymentStatusReport deploymentStatusReport) throws CodePushReportStatusException {
+
+        /* TODO: Consider moving the following logic to some other place or removing it at all if useless. */
+        deploymentStatusReport.setClientUniqueId(mClientUniqueId);
+        deploymentStatusReport.setDeploymentKey(mDeploymentKey);
+        deploymentStatusReport.setAppVersion(deploymentStatusReport.getLocalPackage() != null ? deploymentStatusReport.getLocalPackage().getAppVersion() : mAppVersion);
+        deploymentStatusReport.setAppVersion(deploymentStatusReport.getLocalPackage() != null ? deploymentStatusReport.getLocalPackage().getLabel() : null);
+        final String requestUrl = mServerUrl + REPORT_DEPLOYMENT_STATUS_ENDPOINT;
+        if (deploymentStatusReport.getLocalPackage() != null) {
```

Comment:
Do we need this condition? Further we are manipulating with `deploymentStatusReport.getStatus()` only.